tothic/tothic.go

```diff
@@ -21,19 +21,38 @@ import (
 // SessionName is the key used to access the session store.
 const SessionName = "_gothic_session"
 
+const EnvPrefix = "TYK_IB"
+
+var log = logrus.New()
+
 var TothErrorHandler func(string, string, error, int, http.ResponseWriter, *http.Request)
 
 // Store can/should be set by applications using gothic. The default is a cookie store.
 var Store sessions.Store
 
 func init() {
-	key := []byte(os.Getenv("SESSION_SECRET"))
-	if string(key) == "" {
-		fmt.Println("toth/tothic: no SESSION_SECRET environment variable is set. The default cookie store is not available and any calls will fail. Ignore this warning if you are using a different store.")
-	}
+	key := KeyFromEnv()
 	Store = sessions.NewCookieStore([]byte(key))
 }
 
+func KeyFromEnv() (key string) {
+	// To handle deprecation
+	key = os.Getenv("SESSION_SECRET")
+	temp := os.Getenv(EnvPrefix + "_SESSION_SECRET")
+	if temp != "" {
+		if key != "" {
```

Comment:
LGTM but can you move this to the L41? as in `if temp != "" && key != ""  {` to avoid nesting if statements.

Will be a great idea if you also add tests for this.